test/DynamoCoreTests/UserDataMigrationTests.cs

```diff
@@ -0,0 +1,198 @@
+
+            fv1 = new FileVersion(-1, 9);
+            fv2 = new FileVersion(0, 0);
+            Assert.IsTrue(fv1 < fv2);
+            Assert.IsTrue(fv2 > fv1);
+
+            fv1 = new FileVersion(-2, 9);
+            fv2 = new FileVersion(-1, 0);
+            Assert.IsTrue(fv1 < fv2);
+            Assert.IsTrue(fv2 > fv1);
+
+            fv1 = new FileVersion(2, -9);
+            fv2 = new FileVersion(3, 0);
+            Assert.IsTrue(fv1 < fv2);
+            Assert.IsTrue(fv2 > fv1);
+
+            fv1 = new FileVersion(2, -9);
+            fv2 = new FileVersion(2, -8);
+            Assert.IsTrue(fv1 < fv2);
+            Assert.IsTrue(fv2 > fv1);
+
+            fv1 = new FileVersion(-2, -9);
+            fv2 = new FileVersion(-2, -8);
+            Assert.IsTrue(fv1 < fv2);
+            Assert.IsTrue(fv2 > fv1);
+
+            fv1 = new FileVersion(-2, -9);
+            fv2 = new FileVersion(-1, -9);
+            Assert.IsTrue(fv1 < fv2);
+            Assert.IsTrue(fv2 > fv1);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void SortFileVersionList_SortedList()
+        {
+            var list = new List<FileVersion>();
+
+            var fv1 = new FileVersion(0, 7);
+            var fv2 = new FileVersion(0, 8);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            fv1 = new FileVersion(0, 0);
+            fv2 = new FileVersion(0, 0);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            
+            fv1 = new FileVersion(1, 7);
+            fv2 = new FileVersion(1, 8);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            fv1 = new FileVersion(1, 9);
+            fv2 = new FileVersion(2, 0);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            fv1 = new FileVersion(-1, 9);
+            fv2 = new FileVersion(0, 0);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            fv1 = new FileVersion(-2, 9);
+            fv2 = new FileVersion(-1, 0);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            fv1 = new FileVersion(2, -9);
+            fv2 = new FileVersion(3, 0);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            fv1 = new FileVersion(2, -9);
+            fv2 = new FileVersion(2, -8);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            fv1 = new FileVersion(-2, -9);
+            fv2 = new FileVersion(-2, -8);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            fv1 = new FileVersion(-2, -9);
+            fv2 = new FileVersion(-1, -9);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            list.Sort();
+
+            VerifySortedOrder(list);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void GetSortedInstalledVersions_FromDirectoryInspection()
+        {
+            var tempPath = Path.GetTempPath();
+            var uniqueDirectory = Directory.CreateDirectory(Path.Combine(tempPath, "0.7"));
```

Comment:
Uh oh, wrong. You didn't read my comment properly. I need this to be `UnitTestBase.TempFolder` because that is a temporary folder which is guaranteed to be unique, and will be destroyed after the test.